plugins/main/vlan/vlan.go

```diff
@@ -127,6 +159,10 @@ func cmdAdd(args *skel.CmdArgs) error {
 	}
 	defer netns.Close()
 
+	if n.MasterInContainer != false {
+		args.IfName = getIfName(n.Master, n.VlanId)
+	}
```

Comment:
We shouldn't be doing this here or in cmdDel().  CNI requires that the plugin use the name passed as CNI_IFNAME, and if that's eth0 then the vlan interface must be called eth0.  We can't override it from the plugin per the spec.